src/components/Header.tsx

```diff
@@ -31,10 +56,19 @@ export default function Header() {
           <div className="relative w-max space-x-14 flex top-2">
             {headerSubSections.map((section, index) => (
               <button
-                onMouseEnter={handleMouseEnter}
-                onMouseLeave={handleMouseLeave}
+                onMouseEnter={() => {
+                  handleMouseEnter();
+                  changeHeaderContent(section.text);
+                }}
+                onMouseLeave={() => {
+                  handleMouseLeave();
+                  changeHeaderContent(section.text);
```

Comment:
Why does this get set again on leave? That's kind of a curious implementation

Reply:
addressed in different branch.